kong/core/router.lua

```diff
@@ -446,25 +446,24 @@ function _M.new(apis)
   end)
 
 
-  local grab_headers = #wildcard_hosts > 0 or next(indexes.plain_hosts) ~= nil
+  local grab_host = #wildcard_hosts > 0 or next(indexes.plain_hosts) ~= nil
 
 
-  local function find_api(method, uri, headers)
+  local function find_api(method, uri, host)
     if type(method) ~= "string" then
       return error("arg #1 method must be a string")
     end
     if type(uri) ~= "string" then
       return error("arg #2 uri must be a string")
     end
-    if type(headers) ~= "table" then
-      return error("arg #3 headers must be a table")
+    if host and type(host) ~= "string" then
```

Comment:
Actually on second thought, it is better as it is :) The `~= "string"` is key here